Forge.TreeWalker/src/TreeSchemaValidator.cs

```diff
@@ -0,0 +1,275 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TreeSchemaValidator.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The TreeSchemaValidator class implements the ITreeSchemaValidator interface.
+// </summary>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Forge.DataContracts;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+
+namespace Microsoft.Forge.TreeWalker
+{
+    public class TreeSchemaValidator : ITreeSchemaValidator
+    {
+        private JSchema ForgeSchemaValidationRules;
+        public TreeSchemaValidator() => ForgeSchemaValidationRules = GetRulesFromPath("contracts\\ForgeSchemaValidationRules.json");
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchema(ForgeTree schema)
+        {
+            return await Task.Run(() => Validate(new List<object> { schema }, ForgeSchemaValidationRules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemas(IList<ForgeTree> schemas)
+        {
+            return await Task.Run(() => Validate(ConvertSchemasToSchemaList(schemas), ForgeSchemaValidationRules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemaString(string schema)
+        {
+            return await Task.Run(() => Validate(new List<object> { schema }, ForgeSchemaValidationRules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemasString(string schemas)
+        {
+            return await Task.Run(() => Validate(new List<object> { schemas }, ForgeSchemaValidationRules));
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchemaInPath(string path)
+        {
+            return await GetschemaFromPathAndValidate(path, ForgeSchemaValidationRules);
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateMultipleSchemasInPath(string path)
+        {
+            return await GetAllSchemasInDirectoryAndValidate(path, ForgeSchemaValidationRules);
+        }
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchema(ForgeTree schema, string rules)
```

Comment:
>string rules [](http://example.com/codeflow?start=87&length=12)

We also want to support the case of multiple rules files that link to each other. See the below code as an example.

`
JObject schema = JObject.Parse(this.schemaJSON);

JSchemaPreloadedResolver resolver = new JSchemaPreloadedResolver();
resolver.Add(new Uri("//ForgeSchemaValidationRules.json"), this.forgeRulesJSON);
JSchema rules = JSchema.Parse(this.myAppForgeRulesJSON, resolver);

IList<string> res;
bool isValid = schema.IsValid(rules, out res);
`

Inside the the custom "MyAppForgeSchemaValidationRules", it references the original ForgeSchemaValidationRules like so:

`
"allOf": [
  {
    "$ref": "[file://ForgeSchemaValidationRules.json#/definitions/ActionDefinition](file://ForgeSchemaValidationRules.json#/definitions/ActionDefinition)"
  },
`

Perhaps we could have a method that creates the linked JSchema rules. That way, users can call 2 methods and we don't have to create a ton of overloaded methods.

public static JSchema GetLinkedJSchema(string childRules, string parentRules, string referenceUri)